From 0e576f796c4e07ba8457c1d39ea7e145d5641aaa Mon Sep 17 00:00:00 2001
From: Tae Park <tae.park@windriver.com>
Date: Tue, 9 Sep 2025 14:59:42 +0000
Subject: [PATCH] Add trusted CA Cert secret

Adding a secret that contains a trusted CA cert for portieris. It allows
communication with the image trust server.

Signed-off-by: Tae Park <tae.park@windriver.com>
---
 helm/portieris/templates/deployment.yaml | 11 +++++++++++
 helm/portieris/templates/secret.yaml     | 11 +++++++++++
 helm/portieris/values.yaml               |  6 ++++++
 3 files changed, 28 insertions(+)

diff --git a/helm/portieris/templates/deployment.yaml b/helm/portieris/templates/deployment.yaml
index 826fdba..53d120f 100644
--- a/helm/portieris/templates/deployment.yaml
+++ b/helm/portieris/templates/deployment.yaml
@@ -51,6 +51,12 @@ spec:
           - name: portieris-certs
             readOnly: true
             mountPath: "/etc/certs"
+          {{- if .Values.TrustedCACert }}
+          - name: trusted-cert
+            readOnly: true
+            mountPath: /etc/pki/tls/certs/trustedcert.pem
+            subPath: trustedcert.pem
+          {{- end }}
           livenessProbe:
             httpGet:
               port: 8000
@@ -86,3 +92,8 @@ spec:
       - name: portieris-certs
         secret:
           secretName: portieris-certs
+      {{- if .Values.TrustedCACert }}
+      - name: trusted-cert
+        secret:
+          secretName: trusted-cert
+      {{- end }}
diff --git a/helm/portieris/templates/secret.yaml b/helm/portieris/templates/secret.yaml
index 805b7d3..e8c6c46 100644
--- a/helm/portieris/templates/secret.yaml
+++ b/helm/portieris/templates/secret.yaml
@@ -39,3 +39,14 @@ data:
   {{- end }}
 {{ end }}
 {{ end }}
+{{ if .Values.TrustedCACert }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: trusted-cert
+  namespace: {{ .Release.Namespace }}
+type: Opaque
+data:
+  trustedcert.pem: {{ .Values.TrustedCACert | quote }}
+{{ end }}
diff --git a/helm/portieris/values.yaml b/helm/portieris/values.yaml
index ebe26e5..c9ff82f 100644
--- a/helm/portieris/values.yaml
+++ b/helm/portieris/values.yaml
@@ -53,6 +53,12 @@ UseGeneratedCerts:
   tlsKey: |-
   caCert: |-
 
+# Add the specified certificate(s) to the pod filesystem trust store.
+# The golang x509 module will read all files in this location, and
+# portieris will trust those certificates.
+# The value must be a base 64 encoded list of certificate(s) in PEM format.
+TrustedCACert: ""
+
 # Resoures defined to assist scheduling
 # request is typical x10, limit is typical x100
 resources:
-- 
2.34.1

